app/server/utils/metrics/filters.py

```diff
@@ -19,24 +36,26 @@ def apply_filters(query, filters, query_table):
 
     for filter_table_name, _filts in filters.items():
         if filter_table_name == TransferAccount.__tablename__:
-                # only join transfer account if table is not transfer account
-                if query_table.__tablename__ == filter_table_name: 
-                    query = _apply_single_column_filter(query, _filts, TransferAccount, None, None) 
-                else:
-                    query = _apply_single_column_filter(query, _filts, TransferAccount, account_join_attribute, None)
```

Comment:
pulled out the last two terms and this became a bit simpler

Reply:
This broke the search tests (which also use metrics filters) so I had to roll some of these changes back